CIP-XXXX/README.md

```diff
@@ -0,0 +1,235 @@
+
+The metadata envelope has the following basic structure:
+
+![x509 Secure Metadata Envelope structure](images/metadata-envelope-structure.svg)
+
+The purpose of this data is to lock a particular set of Auxiliary data verifiably to its original transaction.
+It cannot be permitted to take the Auxiliary data to another transaction and just attach it without it being detectable.
+
+Because Plutus scripts cannot access the Auxiliary Data of a transaction,
+this detection can NOT be done on-chain by ledger rules or Plutus scripts.
+This metadata is designed to be processed by off-chain processors of this information.
+
+### Detailed description of the fields in the metadata and their purpose
+
+The metadata is encoded as a CBOR map, and follows the [specification for the encoding of all Cardano metadata].
+
+#### Key: 0 = Purpose
+
+Purpose is defined by the consuming dApp.
+It allows a dApp to have their own privately named and managed namespace for certificates.
+The x509 specifications presented here do not define how certificates must be created.
+Nor the purpose they are used for, that is within control of each dApp.
+These specifications define a universal framework to implement such systems from.
+
+For each set of "Purposes" that a dApp has for role-based access or off-chain identity to on-chain identity management
+the dApp will define a V4 UUID.
+There is no central repository of these, though a dApp may publish the ID it uses and what it's used for.
+The reason UUID V4 is chosen here is if properly chosen, they can be self-assigned without fear of collision
+with another dApp and eliminate the need for a central registry to manage them.
+
+A dApp can define multiple purposes to segment RBAC functions by class.
+For example, Project Catalyst defines the following two Purposes:
+
+| UUID-V4 | Purpose |
+| --- | --- |
+| ca7a1457-ef9f-4c7f-9c74-7f8c4a4cfa6c | Project Catalyst User Role Registrations |
+| ca7ad312-a19b-4412-ad53-2a36fb14e2e5 | Project Catalyst Admin Role Registrations |
+
+The reason in this case for having two purposes for the same application is to allow the rules
+governing registration to be easier managed between the groups.
+It also allows Admins to have a distinct identity when they are just users of the system.
+Any dApp is free to have as many "purposes" defined as suits their goals.
+
+#### Key 1: txn-inputs-hash
+
+This is a hash of the transaction inputs field from the transaction this auxiliary data was originally attached to.
+This hash anchors the auxiliary data to a particular transaction, if the auxiliary data is moved to a new transaction
+it is impossible for the txn-inputs-hash to remain the same, as UTXO's can only be spent once.
+
+This is a key to preventing replayability of the metadata, but is not enough by itself as it needs to be able to be
+made immutable, such that any change to this field can be detected.
```

Comment:
Need to provide more explicit context for exactly how the hash of the inputs will be determined and then matched against the transaction in question to validate and verify.

* What type of hash is used (not otherwise specified)?
* How must one assemble the txn-inputs in order to create a canonical package suitable for hashing and validating?